engines/sci/graphics/text32.cpp

```diff
@@ -378,12 +588,8 @@ int16 GfxText32::Size(Common::Rect &rect, const char *text, GuiResourceId fontId
 		rect.right = maxWidth ? maxWidth : MIN(rect.right, maxTextWidth);
 	}
 
-	// Adjust the width/height if we're using an upscaled font
-	// for the scripts
-	if (_screen->fontIsUpscaled()) {
-		rect.right = rect.right * _screen->getWidth() / _screen->getDisplayWidth();
-		rect.bottom = rect.bottom * _screen->getHeight() / _screen->getDisplayHeight();
-	}
+	rect.right = rect.right * scriptWidth / _scaledWidth;
+	rect.bottom = rect.bottom * scriptHeight / _scaledHeight;
```

Comment:
From my limited SCI knowledge I know that the original used different Rect styles (inclusive originally vs exclusive for us). When we do scaling like this, the resulting rect grows slower for downscaling and faster for upscaling in our scheme (AFAICT right now). So, it might be worth considering to use Sierra style scaling and then patch up coordiantes.